### PR TITLE
added recipe for semi-ate-spyder

### PR DESCRIPTION
**Pull request by nerohmot**
Checklist
- [X] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [X] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [X] Source is from official source.
- [X] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [X] If static libraries are linked in, the license of the static library is packaged.
- [X] Build number is 0.
- [X] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [X] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [X] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by nerohmot** [comment]
@chrisburr could you merge ?

**Comment by chrisburr** [comment]
As I mentioned in https://github.com/conda-forge/staged-recipes/pull/18852#issuecomment-1118426485:

> Please [request reviews from the staged recipes team](https://github.com/conda-forge/staged-recipes#12-my-pull-request-passes-all-checks-but-hasnt-received-any-attention-how-do-i-call-attention-to-my-pr--what-is-the-customary-amount-of-time-to-wait) rather than pinging users directly.

File: recipes/semi-ate-spyder/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+    - pydantic
+    - python >=3.8
+    - qtawesome
+    - semi-ate-common
+    - semi-ate-plugins
+    - semi-ate-project-database
+    - semi-ate-sammy
+    - spyder >=5.3
+    - numpydoc ==1.3.1
+#    - applaunchservices >=0.1.7
```
Comment (goanpeca) [inline recipes/semi-ate-spyder/meta.yaml:33]: Could you remove this comment?

File: recipes/semi-ate-spyder/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+    
+    
+test:
+  imports:
+    - ate_spyder
+    - ate_spyder.widgets
+#  commands:
+#    - pip check
+#  requires:
+#    - pip
```
Comment (goanpeca) [inline recipes/semi-ate-spyder/meta.yaml:44]: Could you remove these comments?

**Comment by nerohmot** [comment]
@goanpeca
Hi Gonzalo,
if I add the `pip check` again then OSX starts failing 
image
with as a trace :
image
`applauncherservices` is introduced by `spyder`, and as `spyder` is ofcourse a dependency of this spyder-plugin ...
image
as a compromise, is it acceptable that I do the same as the `spyder` recipe ie : `pip check  # [linux or win]` ?
image

File: recipes/semi-ate-spyder/meta.yaml (hunk 1 above)
Comment (nerohmot) [reply recipes/semi-ate-spyder/meta.yaml]: ok

File: recipes/semi-ate-spyder/meta.yaml (hunk 2 above)
Comment (nerohmot) [reply recipes/semi-ate-spyder/meta.yaml]: :-)

**Comment by nerohmot** [comment]
I also asked @ccordoba12 for a 'cleaner' solution from spyder side in spyder-ide/spyder#17835, but I would appreciate it if we can proceed as proposed, and not have to wait for the cleaner solution :innocent:

**Comment by BastianZim** [comment]
Since it’s noarch it will only be built on Linux in the feedstock and we can ignore the OSX failure here.

File: recipes/semi-ate-spyder/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+    - numpy
+    - pydantic
+    - python >=3.8
+    - qtawesome
+    - semi-ate-common
+    - semi-ate-plugins
+    - semi-ate-project-database
+    - semi-ate-sammy
+    - spyder >=5.3
+    - numpydoc ==1.3.1
```
Comment (carterbox) [inline recipes/semi-ate-spyder/meta.yaml:32]:
```suggestion
```
This is not listed in the requirements upstream.

**Review by carterbox — CHANGES_REQUESTED** [review]

**Comment by nerohmot** [comment]
@carterbox, you would think so, however when applying your suggestion I get :
image
With as win_64 trace: 
image
and as trace for linux_64 :
image
So before going down the `jinja2` rabbit hole, I first tried to force `numpydoc` to the latest version (hence the `=1.3.1`) and luckily for me conda found a solution for that!!! :sweat_smile:

File: recipes/semi-ate-spyder/meta.yaml (hunk 2 above)
Comment (goanpeca) [reply recipes/semi-ate-spyder/meta.yaml]: I meant delete the lines, just to make the recipe cleaner ;-) 

pip check does not work all the time, due to package name differences or other issues.

File: recipes/semi-ate-spyder/meta.yaml (hunk 1 above)
Comment (goanpeca) [reply recipes/semi-ate-spyder/meta.yaml]: I also meant to delete it if it was not going to be used

**Comment by nerohmot** [comment]
@conda-forge/staged-recipes Can we merge now?

**Comment by carterbox** [comment]
> numpydoc 1.2.1 has requirement Jinja2<3.1,>=2.10, but you have jinja2 3.1.2.

This error does not say that your package requires numpydoc. It says that numpydoc was installed in the test environment and the numpydoc=1.2.1 package is broken.

File: recipes/semi-ate-spyder/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+    
+test:
+  imports:
+    - ate_spyder
+    - ate_spyder.widgets
+  commands:
+    - pip check  # [linux or win]
+  requires:
+    - pip
```
Comment (carterbox) [inline recipes/semi-ate-spyder/meta.yaml:41]:
```suggestion
    
test:
  imports:
    - ate_spyder
    - ate_spyder.widgets
  commands:
    - pip check  # [linux or win]
  requires:
    - pip
    # This build of numpydoc has broken requirements
    - numpydoc !=1.2.1=*_0
```
Something like this, so the test environment doesn't choose the broken numpydoc build.

**Comment by carterbox** [comment]
Also, you don't need to take screenshots. If you want to quote relevant text, just copy the log text directly.

**Review by carterbox — CHANGES_REQUESTED** [review]

**Comment by carterbox** [comment]
There's probably a similar issue with applaunchservices. That package incorrectly specified their dependencies / didn't do pip check in their feedstock, so the pip check error shows up here. Figure out which build of applaunchservices isn't broken and add that as a constraint to your test environment.

File: recipes/semi-ate-spyder/meta.yaml (hunk 4 above)
Comment (nerohmot) [reply recipes/semi-ate-spyder/meta.yaml]: I see

**Pull request merged**